pubsub/rabbitmq/rabbitmq.go

```diff
@@ -190,10 +193,22 @@ func (r *rabbitMQ) publishSync(req *pubsub.PublishRequest) (rabbitMQChannelBroke
 		routingKey = val
 	}
 
+	ttl, ok, err := contribMetadata.TryGetTTL(req.Metadata)
+	if err != nil {
+		r.logger.Errorf("%s publishing to %s failed in TryGetTTL: %v", logMessagePrefix, req.Topic, err)
+		return r.channel, r.connectionCount, err
+	}
+	var expiration string
+	if ok {
+		// RabbitMQ expects the duration in ms
+		expiration = strconv.FormatInt(ttl.Milliseconds(), 10)
+	}
```

Comment:
There should be an `else` clause that sets `expiration` to `defaultQueueTTL` from the component metadata!

If there was no TTL specified on the request we should use the one on from the component metadata if that was specified. Only if neither were specified to we not set any expiration.

Reply:
Thanks, this is a good point. In fact, this is related to [rabbitmq itself](https://www.rabbitmq.com/ttl.html):
A TTL can be specified on a  queue or per-message basis. When both a per-queue and a per-message TTL are specified, the lower value between the two will be chosen.

TTL from component metadata is set to rabbitmq queue, and here need not to handle it.

Reply:
After a review of this, rabbitmq will use the lower value when both are set. I will change it to what you suggested, in this way, TTL specified on the request takes priority over TTL specified on the component metadata.

Reply:
Done.